File: 1004 Max Consecutive Ones III/1004.java
class Solution {
    public int longestOnes(int[] A, int K) {
        if (A == null || A.length == 0) {
            return 0;
        }
        int zeros = 0;
        int res = 0;
        int left = 0, right = 0;
        while (right < A.length) {
            if (A[right] == 0) {
                zeros++;
            } 
            
            while (zeros > K) {
                zeros = A[left++] == 0 ? zeros - 1: zeros;
            }
            right++;
            res = Math.max(res, right - left);
        }
        return res;
    }

    // Time Complexity: O(n)
    // Space Complexity: O(1)
}

File: 1004 Max Consecutive Ones III/1004.py
class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        res = 0
        zeros = 0
        left, right = 0, 0
        while right < len(A):
            if A[right] == 0:
                zeros += 1
            while zeros > K:
                zeros = zeros - 1 if A[left] == 0 else zeros
                left += 1
            right += 1
            res = max(res, right - left)
        return res